src/components/scene/ar-hit-test.js

```diff
@@ -1,5 +1,29 @@
+var register = require('../../core/component').registerComponent;
+var THREE = require('../../lib/three');
+var CAM_LAYER = 21;
+var previousFrameAnchors = new Set();
+var anchorToObject3D = new Map();
+var hitTestCache;
+var tempVec3 = new THREE.Vector3();
+var tempQuaternion = new THREE.Quaternion();
+var fakePose = {
+  transform: {
+    orientation: new THREE.Quaternion(),
+    position: new THREE.Vector3()
+  }
+};
+
+function applyPose (pose, object3D, offset) {
```

Comment:
maybe tmp variables can be wrapped in closure?

var applyPose = (function () {
  var tempVec3 = new THREE.Vector3();
  var tempQuaternion = new THREE.Quaternion();
  return function (pose, object3D, offset) {
    ....
  }
})()

Reply:
I think I am also using some of the temp variables elsewhere but this would be neater, i'll figure something out. Maybe just have 2 sets of temp variables for each use case.